addons/payment_mollie/models/payment_transaction.py

```diff
@@ -58,12 +58,13 @@ def _mollie_prepare_payment_request_payload(self):
         base_url = self.acquirer_id.get_base_url()
         redirect_url = urls.url_join(base_url, MollieController._return_url)
         webhook_url = urls.url_join(base_url, MollieController._notify_url)
+        decimal_places = 0 if self.currency_id.name in ('ISK', 'JPY') else 2
```

Comment:
```suggestion
        # See https://docs.mollie.com/docs/multicurrency
        decimal_places = 0 if self.currency_id.name in ('ISK', 'JPY') else 2
```

Reply:
Why don't we use the currency's decimal places set in Odoo?

Reply:
Good idea, I used the ones defined in the payment constants :) (unfortunately the ones in res_currency aren't compliant with Mollie)

Reply:
Don't use the currency decimal places, we had to revert https://github.com/odoo/odoo/pull/119308/files#diff-ddeebeceaf8b5a8ba7f4d1899d31da53ffe9ed3809c2b801429f189427396c16 to rely again on the constant, and not the decimal places. I don't know if it's possible in 15.0, but people can change the precision of a currency, which then sends the wrong amount to the providers

Reply:
Yep, I'm relying on `CURRENCY_MINOR_UNITS` which is a constant (not on the data in `res_currency`) :)

Reply:
Yes, which is why I didn't r- =P
I was answering Valentin more than anything ^^